app/jobs/parole_data_import_job.rb

```diff
@@ -0,0 +1,62 @@
+require 'net/imap'
+require 'mail'
+require 'csv'
+
+class ParoleDataImportJob < ApplicationJob
+  queue_as :default
+
+  def perform(date)
+    @date = date
+    imap = Net::IMAP.new('imap.gmail.com', 993, true)
+    mail = Mail.new(fetch_email(imap))
+    mail.attachments.empty? ? Rails.logger.info('No attachments found') : process_attachments(mail)
+    imap.logout
+    imap.disconnect
+  end
```

Comment:
This should probably be in an ensure block so errors ensure logout/disconnect:
```suggestion
  ensure
    if imap
      imap.logout
      imap.disconnect
    end
  end
```